### DISCUSSION TITLE
schema中type和组件值的对应关系

### DISCUSSION THREAD

**Discussion by MatriXiao88** [alibaba/formily#1646] [Q&A 2.x] [answered]
如下图，声明中设置的type是string，但value值却是number类型，看起来类型不一致，但又能正常工作
问题1：部分demo代码中，date组件对应的type也是声明为string，那 `SchemaTypes.date` 类型的作用是啥，又是啥时候去使用呢
问题2：能否文档中给出关于type和x-component这一部分对应关系的说明，或者给出一个映射表？

我自己在formily/antd的页面上看了那些demo，看起来值类型和type对应的demo也不少，但都能正常工作，就很奇怪

![image](https://user-images.githubusercontent.com/13702561/123356904-3160b280-d59b-11eb-9f46-e75300a94aee.png)

**Comment by janryWang** [comment, marked as answer]
type和值是弱映射关系，因为这个是无法保证的，x-component指定的组件，它onChange返回的数据到底是啥类型，formily根本不知道，formily唯一能做的就是根据type分配不同的字段模型而已，当然，我们还是倾向于让用户遵守这个类型规范，比如用string，自己就传x-component为Input，而不是NumberPicker

**Reply by MatriXiao88** [reply]
我捋了一下文档，是不是可以这样来映射type和字段模型

1. datetime, date, boolean, number, string, 映射Field模型
2. object, 映射ObjectField模型
3. array, 映射ArrayField模型
4. void, 映射VoidField模型
5. (string & {}), 映射的是啥字段模型呢？

至于最终组件返回的是啥类型的值，则是由组件onChange(value)回调中的value类型决定的是么？

**Reply by janryWang** [reply]
字段模型只会识别array/object/void，其他都是普通Field模型
